src/lib/Model/PublicModel.ts

```diff
@@ -1,149 +1,429 @@
+     * by default: `Paste` 
+     */
     pasteLabel?: string,
+    /**
+     * Alert label in use (e.g. MacOS) if access to the clipboard is denied (just PRO), 
+     * by default: `Use ⌘ + c for copy, ⌘ + x for cut and ⌘ + v for paste.`
+     */
     actionNotSupported?: string;
+    /**
+     *  Alert label in use (e.g. Firefox) if access to the clipboard is denied (just PRO), 
+     * by default: ` Use ctrl + c for copy, ctrl + x for cut and ctrl + v for paste.`
+     */
     appleMobileDeviceContextMenuPasteAlert?: string,
+    /**
+     * Alert label in use (e.g. Firefox) if access to the clipboard is denied (just PRO), 
+     * by default: `This action is not supported in this browser.`
+     */
     otherBrowsersContextMenuPasteAlert?: string,
 }
 
-// ASK ARCHITECT BEFORE INTRODUCING ANY CHANGE!
+/**
+ * Describes your custom cell templates as key-value object
+ * 
+ * @see https://reactgrid.com/docs/3.1/7-api/0-interfaces/6a-cell-templates/
+ */
 export interface CellTemplates {
     [key: string]: CellTemplate;
 }
 
-// ASK ARCHITECT BEFORE INTRODUCING ANY CHANGE!
+/**
+ * Describes focus position inside the grid.
+ * 
+ * @see https://reactgrid.com/docs/3.1/7-api/0-interfaces/6-cell-location/
+ */
 export interface CellLocation {
+    /** Row id of cell location */
     readonly rowId: Id;
+    /** Column id of cell location */
     readonly columnId: Id;
 }
 
-
-// ASK ARCHITECT BEFORE INTRODUCING ANY CHANGE!
+/**
+ * Highlight is an element to mark any cell inside the grid 
+ * 
+ * @see https://reactgrid.com/docs/3.1/7-api/0-interfaces/9-highlight/
+ */
 export interface Highlight {
+    /**  Row id of the cell to highlight */
     readonly rowId: Id;
+    /** Column id of the cell to highlight */
     readonly columnId: Id;
+    /** Optional border color */
     readonly borderColor?: string;
+    /** Optional CSS classname of the highlighted cell */
     readonly className?: string;
 }
 
+/**
+ * Union of basic cells usually used for consuming changes and marking cells array inside the data row
+ * 
+ * @see https://reactgrid.com/docs/3.1/7-api/1-types/6-default-cells/
+ */
+export type DefaultCellTypes =
+    | CheckboxCell
+    | DateCell
+    | EmailCell
+    | ChevronCell
+    | HeaderCell
+    | NumberCell
+    | TextCell
+    | TimeCell
 
-// ASK ARCHITECT BEFORE INTRODUCING ANY CHANGE!
-export type DefaultCellTypes = CheckboxCell | DateCell | EmailCell | ChevronCell | HeaderCell | NumberCell | TextCell | TimeCell;
-
-
-// ASK ARCHITECT BEFORE INTRODUCING ANY CHANGE!
+/**
+ * `CellChange` type is used by `onCellsChanged`. It represents mutually exclusive changes on a single cell.
+ * 
+ * @see https://reactgrid.com/docs/3.1/7-api/1-types/2-cell-change/
+ */
 export type CellChange<TCell extends Cell = DefaultCellTypes & Cell> = TCell extends Cell ? Change<TCell> : never;
 
-// ASK ARCHITECT BEFORE INTRODUCING ANY CHANGE!
+/**
+ * `Change` interface represents a particullar change on a single cell on the cell template.
```

Comment:
* `Change` interface represents a particular change on a single cell on the cell template.